common/check_expired_locked_rules

```diff
@@ -1,56 +1,123 @@
-'''
+"""
 
 import sys
-from rucio.db.sqla.session import get_session
+import traceback
+from collections import defaultdict
+from datetime import datetime, timezone
+
+from rucio.db.sqla import models, session
+from sqlalchemy import and_, null, select, true
+from utils.common import PrometheusPusher
 
 # Exit statuses
 OK, WARNING, CRITICAL, UNKNOWN = 0, 1, 2, 3
 
-
-def main():
-    '''
-    Probe to check the locked expired rules or datasets with locked rules
-    '''
+if __name__ == "__main__":
     status = OK
-    session = get_session()
-    try:
-        query = "select rawtohex(id), scope, name, rse_expression from atlas_rucio.rules where locked=1 and expires_at<sys_extract_utc(localtimestamp)"
-        print 'Locked expired rules'
-        for row in session.execute(query):
-            status = CRITICAL
-            print row[0], row[1], row[2]
-    except Exception as error:
-        print error
-        status = UNKNOWN
-        sys.exit(status)
-    try:
-        query = """select rawtohex(c.id), c.scope, c.name, c.rse_expression from atlas_rucio.rules c,
-                   (select a.scope, a.name from atlas_rucio.dids a
-                   where a.expired_at is not null and a.expired_at < sys_extract_utc(localtimestamp)
-                   and exists (select 1 from atlas_rucio.rules b where a.scope=b.scope and a.name=b.name and locked=1)) d
-                   where c.scope=d.scope and c.name=d.name and locked=1"""
-        print 'Datasets expired with locked rules'
-        for row in session.execute(query):
-            status = CRITICAL
-            print row[0], row[1], row[2], row[3]
-    except Exception as error:
-        print error
-        status = UNKNOWN
-        sys.exit(status)
-    sys.exit(status)
+    session = session.get_session()
 
+    # Select statement used for both metrics
+    base_statement = select(
+        models.ReplicationRule,
+        models.DataIdentifier.name,
+        models.DataIdentifier.scope,
+        models.ReplicationRule.rse_expression,
+    )
+    # Use prometheus pusher to send results to a remote service
+    with PrometheusPusher() as manager:
+        try:
+            statement = base_statement.where(
+                and_(
+                    models.ReplicationRule.locked == true(),
+                    models.ReplicationRule.expires_at < datetime.now(tz=timezone.utc),
+                )
+            )
+            query = session.execute(statement).all()
+            rule_counts = defaultdict(int)
 
-if __name__ == "__main__":
-    main()
+            # Print rules for nagios monitoring
+            print("Locked expired rules")
+            for rule_id, scope, name, rse_expression in query:
+                print(rule_id, scope, name, rse_expression)
+                status = CRITICAL
+                rule_counts[rse_expression] += 1
+
+            # Add an empty entry where there is no result so the metric has continuity
+            if len(rule_counts.keys()) == 0:
```

Comment:
`len(rule_counts) == 0` is better than `len(rule_counts.keys()) == 0`. But if we apply the PEP 8 recommendation for sequences to mappings, then `not rule_counts` is maybe best?